### PR TITLE
EI-1420 add new agreement version and Phase2 eligibility check

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs [hunk 1]
```diff
@@ -179,6 +181,12 @@ public class Phase2Incentive : Incentive
 
         public static int MinimumAgreementVersion() => 6;
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs:182]: are we keeping this?

File: src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs [hunk 2]
```diff
@@ -169,6 +169,8 @@ public class Phase2Incentive : Incentive
         {
         }
 
+        public static int MinimumPhase2AgreementVersion => 6;
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs:172]: why the "Phase2" when it's implied?

Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs]: You can't have a method and a property with the same name, so had to rename it. Changed to use the overloaded method.

File: src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs (hunk 1 above)
Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/Incentive.cs:180]: Yep

**Review by ruslanss — APPROVED** [review]

**Pull request merged**